platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ExportMigrationConfigurationAdminContext.java

```diff
@@ -0,0 +1,171 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.configuration.admin;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.Validate;
+import org.apache.felix.fileinstall.internal.DirectoryWatcher;
+import org.codice.ddf.configuration.persistence.PersistenceStrategy;
+import org.codice.ddf.migration.ExportMigrationContext;
+import org.codice.ddf.migration.ExportMigrationEntry;
+import org.codice.ddf.migration.MigrationEntry;
+import org.codice.ddf.migration.MigrationWarning;
+import org.codice.ddf.migration.ProxyExportMigrationContext;
+import org.osgi.service.cm.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ExportMigrationConfigurationAdminContext extends ProxyExportMigrationContext {
+    private static final Logger LOGGER = LoggerFactory.getLogger(
+            ExportMigrationConfigurationAdminContext.class);
+
+    private final ConfigurationAdminMigratable admin;
+
+    private final Set<String> warnedExtensions = new HashSet<>(8);
+
+    private final Map<Path, ExportMigrationEntry> entries;
+
+    public ExportMigrationConfigurationAdminContext(ExportMigrationContext context,
+            ConfigurationAdminMigratable admin, Configuration[] configs) {
+        super(context);
+        Validate.notNull(admin, "invalid null configuration admin migratable");
+        Validate.notNull(configs, "invalid null configurations");
+        this.admin = admin;
+        this.entries = Stream.of(configs)
+                .map(this::getEntry)
+                .collect(Collectors.toMap(MigrationEntry::getPath, Function.identity()));
+    }
+
+    public Stream<ExportMigrationEntry> entries() {
+        return entries.values()
+                .stream();
+    }
+
+    @Override
+    public ExportMigrationEntry getEntry(Path path) {
+        throw new IllegalStateException("should not be called");
+    }
+
+    @Override
+    public Stream<ExportMigrationEntry> entries(Path path) {
+        throw new IllegalStateException("should not be called");
+    }
+
+    @Override
+    public Stream<ExportMigrationEntry> entries(Path path, PathMatcher filter) {
+        throw new IllegalStateException("should not be called");
+    }
+
+    private ExportMigrationEntry getEntry(Configuration configuration) {
+        Validate.notNull(configuration, "invalid null configuration");
```

Comment:
❓ Instead of checking this here and failing, couldn't we add a `filter` in the stream on L61 to remove `null` configuration objects? Or, even better, to avoid the same filtering logic in both, export _and_ import, we could also filter them out in `ConfigurationAdminMigratable.getConfigurations()`.

Reply:
Actually this is a left over that I should have removed when I was validating args on all method calls. I since then removed those validations from the private methods but forgot these.